react/components/SearchMetadata/components/SearchMetadataModeView/index.js

```diff
@@ -1,52 +1,73 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
 import Pocket from 'react/components/UI/Pocket';
-import HeaderMetadataLinkUnlessCurrent from 'react/components/UI/HeaderMetadata/HeaderMetadataLinkUnlessCurrent';
-
-const SearchMetadataModeView = ({ search }) => {
-  const isCurrent = ({ targetHref, currentRoute }) =>
-    currentRoute.pathname === targetHref.split('?')[0];
-
-  return (
-    <Pocket title="View">
-      <HeaderMetadataLinkUnlessCurrent
-        href={`/search/${search}`}
-        predicate={isCurrent}
-      >
-        All
-      </HeaderMetadataLinkUnlessCurrent>
-
-      <HeaderMetadataLinkUnlessCurrent
-        href={`/search/${search}/channels`}
-        predicate={isCurrent}
-      >
-        Channels
-      </HeaderMetadataLinkUnlessCurrent>
-
-      <HeaderMetadataLinkUnlessCurrent
-        href={`/search/${search}/blocks`}
-        predicate={isCurrent}
-      >
-        Blocks
-      </HeaderMetadataLinkUnlessCurrent>
-
-      <HeaderMetadataLinkUnlessCurrent
-        href={`/search/${search}/users`}
-        predicate={isCurrent}
-      >
-        Users
-      </HeaderMetadataLinkUnlessCurrent>
-    </Pocket>
-  );
-};
-
-SearchMetadataModeView.propTypes = {
-  search: PropTypes.string,
-};
-
-SearchMetadataModeView.defaultProps = {
-  search: '',
-};
-
-export default SearchMetadataModeView;
+import CookieLinkUnlessCurrent from 'react/components/UI/CookieLinkUnlessCurrent';
+
+export default class SearchMetadataView extends Component {
+  static propTypes = {
+    view: PropTypes.oneOf(['all', 'channels', 'blocks', 'users', 'groups']).isRequired,
+    search: PropTypes.string.isRequired,
+  }
+
+  isViewActive = view => () =>
+    this.props.view === view;
+
+  render() {
+    const { search } = this.props;
+
+    return (
+      <Pocket title="View">
+        <CookieLinkUnlessCurrent
+          to={`/search/${search}/all`}
+          isActive={this.isViewActive('all')}
+          prefix="Search"
+          name="view"
+          value="all"
+        >
+          All
+        </CookieLinkUnlessCurrent>
```

Comment:
What was the problem, ultimately?

Reply:
An errant router hoc wrapped on the outer component